pkg/gadgets/trace/tcpdrop/tracer/bpf/tcpdrop.bpf.c

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// Based on tcpdrop(8) from BCC
+//
+// Copyright 2018 Netflix, Inc.
+// 30-May-2018    Brendan Gregg   Created this.
+// 15-Jun-2022    Rong Tao        Add tracepoint:skb:kfree_skb
+// Copyright 2023 Microsoft Corporation
+
+#include <vmlinux/vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+
+#include "tcpdrop.h"
+
+/* Define here, because there are conflicts with include files */
+#define AF_INET		2
+#define AF_INET6	10
+
+// we need this to make sure the compiler doesn't remove our struct
+const struct event *unusedevent __attribute__((unused));
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(__u32));
+	__uint(value_size, sizeof(__u32));
+} events SEC(".maps");
+
+// This struct is defined according to the following format file:
+// /sys/kernel/tracing/events/skb/kfree_skb/format
+struct kfree_skb_ctx {
+	/* The first 8 bytes is not allowed to read */
+	unsigned long pad;
+
+	void *skbaddr;
+	void *location;
+	unsigned short protocol;
+	int reason;
+};
+
+static __always_inline int __trace_tcp_drop(void *ctx, struct sock *sk, struct sk_buff *skb, int reason)
+{
+	if (sk == NULL)
+		return 0;
+
+	u_int8_t *tcphdr = (u_int8_t *)(BPF_CORE_READ(skb, head) + BPF_CORE_READ(skb, transport_header));
+	struct inet_sock *sockp = (struct inet_sock *)sk;
+	struct task_struct *task = (struct task_struct*) bpf_get_current_task();
+
+	struct event event = {};
```

Comment:
Is it really needed to assign it if all fields are filled below?

Reply:
I think it is useful when there is some padding between fields, but I am not sure.

Reply:
I tried to remove this but I get the error:

> Error: running gadget: running gadget: installing tracer: failed to load ebpf program: field IgTcpdrop: program ig_tcpdrop: load program: permission denied: invalid indirect read from stack R4 off -96+8 size 96 (206 line(s) omitted)

Other gadgets have the same pattern:

```
$ git grep 'struct event event = {}'|grep -v Binary
pkg/gadgets/trace/fsslower/tracer/bpf/fsslower.bpf.c:84:	struct event event = {};
pkg/gadgets/trace/open/tracer/bpf/opensnoop.bpf.c:112:	struct event event = {};
pkg/gadgets/trace/signal/tracer/bpf/sigsnoop.bpf.c:40:	struct event event = {};
pkg/gadgets/trace/signal/tracer/bpf/sigsnoop.bpf.c:139:	struct event event = {};
pkg/gadgets/trace/tcp/tracer/bpf/tcptracer.bpf.c:261:	struct event event = {};
pkg/gadgets/trace/tcp/tracer/bpf/tcptracer.bpf.c:300:	struct event event = {};
pkg/gadgets/trace/tcp/tracer/bpf/tcptracer.bpf.c:341:	struct event event = {};
pkg/gadgets/trace/tcpconnect/tracer/bpf/tcpconnect.bpf.c:134:	struct event event = {};
pkg/gadgets/trace/tcpconnect/tracer/bpf/tcpconnect.bpf.c:153:	struct event event = {};
pkg/gadgets/trace/tcpdrop/tracer/bpf/tcpdrop.bpf.c:53:	struct event event = {};
```